### PR TITLE
Add ServerSendChatEvent

### PR DESCRIPTION
**Pull request by shadowfacts**
Allows messages sent from the server (e.g. through `/say`) to be intercepted and canceled/modified.

Tested using [this](https://gist.github.com/shadowfacts/ad6ceffe188f72d1fc16).

### CODE REVIEW

**Comment by williewillus** [comment]
Javadoc on the event

File: patches/minecraft/net/minecraft/server/management/ServerConfigurationManager.java.patch [hunk 1]
```diff
@@ -221,3 +221,14 @@
                  p_82448_4_.func_72838_d(p_82448_1_);
                  p_82448_4_.func_72866_a(p_82448_1_, false);
              }
+@@ -912,7 +971,9 @@
+ 
+     public void func_148539_a(IChatComponent p_148539_1_)
+     {
+-        this.func_148544_a(p_148539_1_, true);
++    	net.minecraftforge.event.ServerSendChatEvent event = new net.minecraftforge.event.ServerSendChatEvent(p_148539_1_);
++    	net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
```
Comment (LexManos) [inline patches/minecraft/net/minecraft/server/management/ServerConfigurationManager.java.patch]: NEVER use tabs

File: patches/minecraft/net/minecraft/server/management/ServerConfigurationManager.java.patch [hunk 2]
```diff
@@ -221,3 +221,14 @@
...
++        if (!event.isCanceled()) this.func_148544_a(event.getMessage(), true);
+     }
```
Comment (LexManos) [inline patches/minecraft/net/minecraft/server/management/ServerConfigurationManager.java.patch]: Also use the event factory to make the patch smaller, there is also no need to move the function, something along the lines of:
p_148539_1_ = Factory.chatEvent(p_148539_1_);
if (p_148539_1_ == null) return;

**Comment by shadowfacts** [comment]
There ya go, switched to spaces and added a helper method.

**Comment by Spartan322** [comment]
Fix those docs to fit Forge event standards, look at other events for reference

**Comment by shadowfacts** [comment]
I rewrote the JavaDoc

**Comment by shadowfacts** [comment]
Closed in favor of #2599 for 1.9.

**Pull request closed without merging**